From: Philip Kuryloski <kuryloskip@vmware.com>
Subject: Fix `rabbitmqctl status` when the disk_free cannot be
 determined

By adjusting RabbitMQ.CLI.InformationUnit to no longer attempt to
convert non-numeric values.

Ubuntu backport note: In rabbitmq 3.8 the convert function is called with a
:undefined value when the disk space cannot be obtained instead of a :unknown
value. Thus the value was changed accordingly with respect to the original
commit.

Origin: backport, https://github.com/rabbitmq/rabbitmq-server/commit/7a5a467d4be4
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1999816
Last-Update: 2023-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex
+++ b/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex
@@ -39,14 +39,18 @@
     :rabbit_resource_monitor_misc.parse_information_unit(val)
   end
 
-  def convert(bytes, "bytes") do
+  def convert(bytes, "bytes") when is_number(bytes) do
     bytes
   end
 
-  def convert(bytes, unit) do
+  def convert(bytes, unit) when is_number(bytes) do
     do_convert(bytes, String.downcase(unit))
   end
 
+  def convert(:undefined, _) do
+    :unknown
+  end
+
   def known_unit?(val) do
     MapSet.member?(known_units(), String.downcase(val))
   end
